### DISCUSSION TITLE
How to compile ONNX Runtime using wasi-sdk instead of using emsdk to wasm?

### DISCUSSION THREAD

**Discussion by chris-void-777** [microsoft/onnxruntime#16999] [Other Q&A]
I want to use onnxruntime in wasm runtime,so I need use wasi-sdk rather than emsdk during the build process. I am not familiar with the build process of onnxruntime.I didn't find the relevant documentation (maybe I missed it, I'm sorry if so), and I checked the build script and some related files (such as ci_build/build.py), but I still don't know how to start. So what should I do to achieve my goal? 

Thanks for your time and help

**Comment by carzh** [comment]
[Here](https://onnxruntime.ai/docs/build/web.html) are the docs we have for building ORT Web from source, which includes building the WASM artifacts. 

However, as far as I know, using wasi-libc instead of emscripten to build the WASM artifacts will be a very difficult process. You'll likely have to edit build.py, cmake/onnxruntime_webassembly.cmake, and onnxruntime/wasm/api.cc as well as api.h... as well as many other files that were written to work with emscripten.

**Reply by chris-void-777** [reply]
Yes, I've read the document, but I didn't find the information I was looking for. It seems that compiling with wasi-sdk is indeed challenging, so it looks like I should consider other solutions. Thanks
